src/ecc_compact/mod.rs

```diff
@@ -121,11 +121,24 @@ impl TryFrom<&[u8]> for PublicKey {
     type Error = Error;
 
     fn try_from(input: &[u8]) -> Result<Self> {
-        match p256::AffinePoint::decompact(&FieldBytes::from_slice(&input[1..])).into() {
-            Some(point) => Ok(PublicKey(
-                p256::PublicKey::from_affine(point).map_err(Error::from)?,
-            )),
-            None => Err(Error::not_compact()),
+        if input.len() == PUBLIC_KEY_LENGTH {
+            // Assume this is a compact key we've encoded before, strip of the network/type tag
+            match p256::AffinePoint::decompact(&FieldBytes::from_slice(&input[1..])).into() {
+                Some(point) => Ok(PublicKey(
+                    p256::PublicKey::from_affine(point).map_err(Error::from)?,
+                )),
+                None => Err(Error::not_compact()),
+            }
+        } else {
+            // Otherwise assume it's just raw bytes
```

Comment:
Is there any possibility that this is correct branch even when `input.len() == PUBLIC_KEY_LENGTH`? If not, then this looks good to me.

Reply:
it _could_ be but if the length > PUBLIC_KEY_LENGTH you want to be _sure_ that bytes represent a fully non-compacted key.. Otherwise it incorrectly makes assumptions about what the second 32 bytes contains